src/vm/mod.rs

```diff
@@ -456,6 +456,11 @@ impl super::VM for CoreVM {
                 handler_name: target.handler,
                 key: target.key.unwrap_or_default(),
                 idempotency_key: target.idempotency_key,
+                headers: target
+                    .headers
+                    .into_iter()
+                    .map(crate::service_protocol::messages::Header::from)
```

Comment:
I think you can do simply `Into::into`

Reply:
i agree, but stylistically i much prefer seeing what type its becoming

Reply:
+1 for what Jack said. `Into::into` is really not helping for readability.